File: samples/client/petstore/java/src/main/java/org/openapijsonschematools/client/schemas/validation/FloatEnumValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;

public interface FloatEnumValidator <EnumType extends FloatValueMethod> {
    float validate(EnumType arg, SchemaConfiguration configuration) throws ValidationException;
}

File: samples/client/3_0_3_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/FloatEnumValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

interface FloatEnumValidator<EnumType : FloatValueMethod> {
    @Throws(ValidationException::class)
    fun validate(arg: EnumType, configuration: SchemaConfiguration?): Float
}